frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -64,7 +72,37 @@ export const experimentLogic = kea<experimentLogicType>({
                     }
                     return { ...vals, ...experimentData }
                 },
-                emptyData: () => null,
+                updateExperimentGroup: (state, { variant, idx }) => {
+                    const featureFlagVariants = [...(state?.feature_flag_variants || [])]
+                    featureFlagVariants[idx] = { ...featureFlagVariants[idx], ...variant }
+                    return { ...state, feature_flag_variants: featureFlagVariants }
+                },
+                addExperimentGroup: (state) => {
+                    if (state?.feature_flag_variants) {
+                        const newRolloutPercentage = Math.round(100 / (state.feature_flag_variants.length + 1))
+                        const updatedRolloutPercentageVariants = state.feature_flag_variants.map((variant) => ({
+                            ...variant,
+                            rollout_percentage: newRolloutPercentage,
+                        }))
+                        return {
+                            ...state,
+                            feature_flag_variants: [
+                                ...updatedRolloutPercentageVariants,
+                                {
+                                    key: `test_group_${state.feature_flag_variants.length + 1}`,
+                                    rollout_percentage: newRolloutPercentage,
+                                },
+                            ],
+                        }
+                    }
+                    return state
+                },
+                emptyNewExperiment: () => ({
```

Comment:
this seems to clash with the default set in the reducer? Do we need the defaults in the `newExperimentData` reducer?

Reply:
yeah probably not since this just overrides the default in the reducer :/